components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -84,16 +87,33 @@ data class NimbusServerSettings(
  */
 class Nimbus(
     context: Context,
-    server: NimbusServerSettings?
-) : NimbusApi {
+    server: NimbusServerSettings?,
+    private val delegate: Observable<Observer> = ObserverRegistry()
+) : NimbusApi, Observable<Nimbus.Observer> by delegate {
```

Comment:
Since we can't have delegation with interfaces, I'm struggling a bit with how the implementation definition looks if the `NimbusApi` derives from `Observable<NimbusObserver>`.

Reply:
I _think_ we should be able to make the interface extend `Observable<NimbusApi.Observer>`, and keeping the delegate on the implementation.

```kotlin
// interface
interface NimbusApi: Observable<NimbusApi.Observer> {
    interface Observer {}
}

// implementation
class Nimbus(
    …
): NimbusApi, Observable<NimbusApi.Observer> by delegate {}

Reply:
Ha, I had that but I had mistakenly broken something somewhere else.  Thanks!